docs/v2/guides/gutenberg.md

```diff
@@ -11,112 +11,123 @@ Timber works with the Block Editor (also called Gutenberg) out of the box. If yo
-- **is_preview** - returns true during AJAX preview
+#### Register the Block
 
-Finally, you can create the template **block/example-block.twig**:
+Your site will need to know the block exists on init so we will employ the [register_block_type](https://developer.wordpress.org/reference/functions/register_block_type/) function witin our functions.php file and trigger the function on init. 
 
-```twig
-{#
-/**
- * Block Name: Example block
- *
- * This is the template that displays the example block.
- */
-#}
-
-{% if is_preview %}
-    <p>I will only appear in the editor.</p>
-{% endif %}
-
-<div id="example-{{ block.id }}" class="wrapper">
-    <h1>{{ fields.title }}</h1>
-    <p>{{ fields.description }}</p>
-</div>
-<style type="text/css">
-    #testimonial-{{ block.id }} {
-        background: {{ fields.background_color }};
-        color: {{ fields.text_color }};
-    }
-</style>
+The straight forward way to do this is posted below: 
+
+```php
+function register_acf_blocks() {
+    register_block_type( __DIR__ . '/blocks/my-block' ); //register_block_type will look in the current directory and register the block you specify
+    //add additional register_block_type functions for each of your custom blocks
+}
+add_action( 'init', 'register_acf_blocks' ); //trigger the register function on init
 ```
 
-If you would like to use an external stylesheet both inside of the block editor and the frontend you should add:
+Alternatively, you can dynamically register each block. This will keep our code DRY and prevent you from having to register each inidivual block every time you create a new one. 
 
 ```php
-function my_acf_block_editor_style()
-{
-    wp_enqueue_style(
-        'example_block_css',
-        get_template_directory_uri() . '/assets/example-block.css'
-    );
+function register_acf_blocks() {
+	foreach ( $blocks = new DirectoryIterator( __DIR__ . '/blocks' ) as $item ) {
+        if ( $item -> isDir() && !$item -> isDot() ) //check for the directory
+		if ( file_exists( $item -> getPathname() . '/block.json' ) ) //check if the block.json exists
+		register_block_type( $item -> getPathname() );  //register the block given the directory name within the blocks directory
+	}
 }
+add_action( 'init', 'register_acf_blocks' ); //trigger the register function on init
+```
+
+#### Create fields in ACF
+
+[ACF has precise guidance](https://www.advancedcustomfields.com/resources/create-your-first-acf-block/#create-the-testimonial-field-group) on how to create the fields for your block. The important thing is to make sure the field group is enabled for your specific block: 
+    Show field group if: Block  
+    Operator: is equal to  
+    Selector: My Block  
+    
+
+#### Block Template
+Now that we have our block directory and settings, we need the block template that our site will render. We will create a blocks directory within our view directory to keep things organized and add the template file to that blocks directory: 
```

Comment:
```suggestion
Now that we have our block directory and settings, we need the a twig template for our block that will be used to display our block. We will create a blocks directory within our view directory to keep things organized and add the template file to that blocks directory: 
```

Reply:
Personally, I have my twig file inside my block directory as well, so everything is one neat package and this seems to be the more WordPress orientated way of organizing blocks. Something that you might want to consider. This does require to register an additional directory to Timber.

![image](https://github.com/timber/timber/assets/17764157/10a50c8c-c1a9-4b5e-9534-7c015ddc476a)

Reply:
ah that's a good idea. I created some of my blocks as components at first so they can be used as field groups for a post or blocks in Gutenberg interchangeably. perhaps for this tutorial it makes things simpler to just add the twig file within the block directory. I had previously thought it all had to be within the views directory.